### PR TITLE
fix #2395 bug with isDirty delete

### LINKED ISSUE #2395: [useFieldArray] form is dirty when append and remove field
**Describe the bug**
When I append a field and then remove it, the `formState.isDirty` is still `true`.

**To Reproduce**
1. Open console
2. Click on append button
3. Remove last field clicking on remove button
4. Check console and isDirty is still `true`

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-usefieldarray-template-ktweo?file=/src/index.js

**Expected behavior**
`formState.isDirty` is `false` at the end of steps

**Additional context**
It doesn't happen in version 6.0.8

### PR DESCRIPTION
**Pull request by bluebill1049**

### CODE REVIEW

File: src/useFieldArray.test.tsx [hunk 1]
```diff
@@ -922,6 +922,64 @@ describe('useFieldArray', () => {
...
+        );
+      };
+
+      render(<Component />);
+
+      expect(formState.isDirty).toBeFalsy();
+
+      await actComponent(async () => {
+        await fireEvent.click(screen.getByRole('button', { name: /append/i }));
+      });
```
Comment (keiya01) [inline src/useFieldArray.test.tsx:970]:
```suggestion
      fireEvent.click(screen.getByRole('button', { name: /append/i }));
```

File: src/useFieldArray.test.tsx [hunk 2]
```diff
@@ -922,6 +922,64 @@ describe('useFieldArray', () => {
...
+
+      expect(formState.isDirty).toBeTruthy();
+
+      await actComponent(async () => {
+        await fireEvent.click(
+          screen.getAllByRole('button', { name: /remove/i })[1],
+        );
+      });
```
Comment (keiya01) [inline src/useFieldArray.test.tsx:978]:
```suggestion
      fireEvent.click(
        screen.getAllByRole('button', { name: /remove/i })[1],
      );
```

**Review by keiya01 — COMMENTED** [review]
I think these method are not async.

**Comment by bluebill1049** [comment]
thanks @keiya01 👍

**Review by keiya01 — DISMISSED** [review]
LGTM :)

**Comment by bluebill1049** [comment]
oops failed lol.

**Comment by keiya01** [comment]
omg ... I think this failure is caused by lint error.

**Comment by bluebill1049** [comment]
lol, fixing it.

**Comment by keiya01** [comment]
Thank you...

**Comment by bluebill1049** [comment]
Thanks very much for the review and feedback @keiya01 🙏

**Pull request merged**